components/browser/engine-gecko-beta/src/main/java/mozilla/components/browser/engine/gecko/prompt/GeckoPromptDelegate.kt

```diff
@@ -0,0 +1,178 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.engine.gecko.prompt
+
+import mozilla.components.browser.engine.gecko.GeckoEngineSession
+import mozilla.components.concept.engine.prompt.Choice
+import mozilla.components.concept.engine.prompt.PromptRequest.MenuChoice
+import mozilla.components.concept.engine.prompt.PromptRequest.MultipleChoice
+import mozilla.components.concept.engine.prompt.PromptRequest.SingleChoice
+import org.mozilla.geckoview.AllowOrDeny
+import org.mozilla.geckoview.GeckoResult
+import org.mozilla.geckoview.GeckoSession
+import org.mozilla.geckoview.GeckoSession.PromptDelegate.AlertCallback
+import org.mozilla.geckoview.GeckoSession.PromptDelegate.AuthCallback
+import org.mozilla.geckoview.GeckoSession.PromptDelegate.AuthOptions
+import org.mozilla.geckoview.GeckoSession.PromptDelegate.ButtonCallback
+import org.mozilla.geckoview.GeckoSession.PromptDelegate.Choice.CHOICE_TYPE_MENU
+import org.mozilla.geckoview.GeckoSession.PromptDelegate.Choice.CHOICE_TYPE_MULTIPLE
+import org.mozilla.geckoview.GeckoSession.PromptDelegate.Choice.CHOICE_TYPE_SINGLE
+import org.mozilla.geckoview.GeckoSession.PromptDelegate.ChoiceCallback
+import org.mozilla.geckoview.GeckoSession.PromptDelegate.FileCallback
+import org.mozilla.geckoview.GeckoSession.PromptDelegate.TextCallback
+
+typealias GeckoChoice = GeckoSession.PromptDelegate.Choice
+
+/**
+ * Gecko-based PromptDelegate implementation.
+ */
+@Suppress("TooManyFunctions")
+class GeckoPromptDelegate(private val geckoEngineSession: GeckoEngineSession) :
```

Comment:
nit: `internal`. This class is not part of the public API.

Reply:
👍